pkg/lightning/backend/local/local.go

```diff
@@ -153,7 +151,32 @@ type File struct {
 	// isImportingAtomic is an atomic variable indicating whether this engine is importing.
 	// This should not be used as a "spin lock" indicator.
 	isImportingAtomic atomic.Uint32
-	mutex             sync.Mutex
+	// flush and ingest sst hold the rlock, other operation hold the wlock.
+	mutex sync.RWMutex
+
+	ctx          context.Context
+	cancel       context.CancelFunc
+	sstDir       string
+	sstMetasChan chan metaOrFlush
```

Comment:
seems the linearization point is when `metaOrFlush` enters this channel, so the order of concurrent calls of `addSST` and `flushEngineWithoutLock` is not guaranteed. Not sure if this will cause problems

Reply:
The consistency guarantee is ensured by the caller. When flushEngineWithoutLock is called, all localwriter belongs to this engine are either closed or flushed, so before sending the flush message, all the related addSST calls are ensured to be done.